pkg/controllers/openshift-controller-manager.go

```diff
@@ -116,13 +119,24 @@ func (s *OCPControllerManager) Run(ctx context.Context, ready chan<- struct{}, s
 	if err := assets.ApplyNamespaces([]string{
 		"assets/core/0000_50_cluster-openshift-controller-manager_00_namespace.yaml",
 	}, s.kubeconfig); err != nil {
-		klog.Warningf("failed to apply openshift namespaces %v", err)
+		klog.Errorf("failed to apply openshift namespaces %v", err)
```

Comment:
Assuming RetryTCPConnection writes an error to the chan, what's the expected sequence of events?  It looks as though the err will be placed in the chan buffer, but ApplyNamespaces will be called.  If ApplyNamespaces errors, then the error written to the chan is lost at the return. 

Or:
If RetryTCPConnection sends an error to the chan buff, but ApplyNamespaces returns nil, then the 2nd go func is run.  Then the select will read from the chan buff and not block since there's a value to read, causing the function to return.  The concurrent StartControllerManager() call will continue running however.  

Question is: is it safe to assume the StartControllerManager routine will die gracefully and quietly (without spamming logs) if RetryTCPConnection returns and error?